sql/src/main/java/io/crate/operation/reference/sys/node/SysNodeStatsExpressions.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.reference.sys.node;
+
+import com.google.common.collect.ImmutableMap;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.RowCollectExpression;
+import io.crate.metadata.RowContextCollectorExpression;
+import io.crate.metadata.sys.SysNodesTableInfo;
+import io.crate.monitor.ExtendedFsStats;
+import io.crate.monitor.ThreadPools;
+import io.crate.operation.reference.sys.node.fs.*;
+import org.apache.lucene.util.BytesRef;
+
+import java.util.Map;
+
+public class SysNodeStatsExpressions {
+
+    private static Map<ColumnIdent, RowCollectExpression> expressions = ImmutableMap.<ColumnIdent, RowCollectExpression>builder()
+        .put(SysNodesTableInfo.Columns.ID, new RowContextCollectorExpression<NodeStatsContext, BytesRef>() {
+            @Override
+            public BytesRef value() {
+                return row.id();
```

Comment:
I think the expressions here are now basically singletons but not thread-safe which will cause issues with concurrent queries?

The caching layer won't help because the expression instance is cached and not the values